src/Form.jsx

```diff
@@ -1,18 +1,13 @@
-var React = require('react');
-var ReactDOM = require('react-dom');
-
-var Fields = require('./fields');
-var fieldType = Fields.fieldType;
-
-var Form = React.createClass({
-  propTypes: {
-    fields: React.PropTypes.objectOf(fieldType).isRequired,
-    onProgress: React.PropTypes.func,
-    onUpdate: React.PropTypes.func
-  },
+import React from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+import Fields, { fieldType } from './fields';
 
+class Form extends React.Component {
   // boilerplate
-  getInitialState: function() {
+  constructor(props) {
+    super(props);
+
     var initial = {};
```

Comment:
we can keep a function that generates this state, so that we can do `this.reset(props = {})` which then has `this.state = {.....}` in it